src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs

```diff
@@ -102,6 +112,27 @@ INamedTypeSymbol dependencyAttributeType
 				// other things to check:
 				// - appropriate use of [Dependency]
 			}
+
+			CheckThatRpcParametersAreValid( context, method, deserializerType );
+		}
+
+		private static void CheckThatRpcParametersAreValid(
+			SyntaxNodeAnalysisContext context,
+			MethodDeclarationSyntax method,
+			INamedTypeSymbol deserializerType
+		) {
+			if( deserializerType.GetType() == null ) {

```

Comment:
Hm, so `serializerType.GetType()` is a call to `System.GetType()` which will always return `typeof( INamedType )` symbol here (never `null`).
```suggestion
			if( deserializerType.GetType() == null ) {
```suggestion
			if( deserializerType.GetType() == null ) {

```
```

There is a distinction between the reflection APIs like `GetType()` (which operate on runtime objects) and the Roslyn APIs like `INamedTypeSymbol`, which refer to things from the "semantic model" of the assembly being compiled.

Based on the context I'm guessing this should be

```suggestion
			if( deserializerType == null ) {
```

but `throw new Exception` seems wrong? Like this would crash in VS if they were missing this type. If you want to fail the build we should emit an error diagnostic rather than crash.

Reply:
Hm, but you already checked for `== null` in the initializer and if you couldn't find this type you bailed, so I think you can just exclude this check entirely.

Reply:
I don't think `IDeserializer` is a _necessary_ type for these general checks. I expect the null check should be removed from the initializer, and then logic surrounding `IDeserializer` should only be considered if the type available (similar to the dependency attribute: https://github.com/Brightspace/D2L.CodeStyle/pull/753/files#diff-459cb71828994b9c631d770281ec937c431c9691462e7dbfa168c7321aa445f3R105)